### DISCUSSION TITLE
Add ability to paste in a beatmap link into beatmap browser/search

### DISCUSSION THREAD

**Discussion by bytenekos** [ppy/osu#30426] [Ideas]
Kinda like how you can do this in youtube:

![image](https://github.com/user-attachments/assets/155e75da-af76-481d-adad-028a3708fefd)

![image](https://github.com/user-attachments/assets/52d561c4-f020-47ec-bb83-9423463a8cb2)

Would be nice for cases where you have a beatmap link (if you're using a discord bot with friends and you wanna quickly and easily play the map) but you don't want to search for it specifically and don't want to open a browser to download from there. 
You could just paste in the link into the beatmap browser and it'd have that link as either the only result or as the 1st beatmap that shows with other maps with the same name shown next

Seen [this discussion ](https://github.com/ppy/osu/discussions/16076) and while yes, you can just use the id, I feel like this would be simpler to the average user (I for example had no idea you had to do this specifically)

**Comment by peppy** [comment]
In stable we used to allow dragging links into the window. I think we should support this.

Optionally we could also read the clipboard when the window comes active and show a notification prompting the user to open the beatmap in the in-game interface. But this should be behind a privacy setting – at very least on iOS this would cause a system level popup due to attempting to access the clipboard.
